mutate/mutate.go

```diff
@@ -298,6 +302,27 @@ func (m *Mutator) Add(ctx context.Context, mediaType string, r io.Reader, histor
 	return desc, nil
 }
 
+// AddExisting adds a blob that already exists to the layer, using the user
+// specified DiffID. It currently checks that the layer exists, but does not
+// validate the DiffID.
+func (m *Mutator) AddExisting(ctx context.Context, desc ispec.Descriptor, history *ispec.History, diffID digest.Digest) error {
+	if err := m.cache(ctx); err != nil {
+		return errors.Wrap(err, "getting cache failed")
+	}
+
+	// make sure the blob exists
+	blob, err := m.engine.GetBlob(ctx, desc.Digest)
+	if err != nil {
+		return err
+	}
+	blob.Close()
```

Comment:
Need to check this error. But in line with #373 we should probably just have a stat-like API so that you don't need to read the whole blob if you're just doing an existence check.

Reply:
Can this wait a bit while I fix #373 so you can build on top of it?

Reply:
#375 has `StatBlob` which is probably want you want to do (actually maybe we don't even need to check if the blob exists?).